pkg/query/api/v1.go

```diff
@@ -626,3 +637,54 @@ func (api *API) labelNames(r *http.Request) (interface{}, []error, *ApiError) {
 
 	return names, warnings, nil
 }
+
+func (api *API) rules(r *http.Request) (interface{}, []error, *ApiError) {
+	var (
+		res       = &storepb.RuleGroups{}
+		typeParam = strings.ToLower(r.URL.Query().Get("type"))
+	)
+
+	if typeParam != "" && typeParam != "alert" && typeParam != "record" {
+		return nil, nil, &ApiError{errorBadData, errors.Errorf("invalid query parameter type='%v'", typeParam)}
+	}
+
+	returnAlerts := typeParam == "" || typeParam == "alert"
+	returnRecording := typeParam == "" || typeParam == "record"
+
+	groups, warnings, err := api.rulesRetriever.RuleGroups(r.Context())
+	if err != nil {
+		return nil, nil, &ApiError{ErrorInternal, fmt.Errorf("error retrieving rules: %v", err)}
+	}
+
+	for _, grp := range groups {
+		apiRuleGroup := &storepb.RuleGroup{
+			Name:                              grp.Name,
+			File:                              grp.File,
+			Interval:                          grp.Interval,
+			EvaluationDurationSeconds:         grp.EvaluationDurationSeconds,
+			LastEvaluation:                    grp.LastEvaluation,
+			DeprecatedPartialResponseStrategy: grp.DeprecatedPartialResponseStrategy,
+			PartialResponseStrategy:           grp.PartialResponseStrategy,
+		}
+
```

Comment:
```suggestion
apiRuleGroup.Rules = make(,len(..))
```
Consider pereallocating rule slice -> https://www.bwplotka.dev/2020/how-thanos-would-program-in-go/#pre-allocating-slices-and-maps

Reply:
agree, that's a good one :+1: